lib/index.ts

```diff
@@ -50,12 +51,70 @@ function analyzeDynamically(
   });
 }
 
-async function analyzeStatically(
-  targetImage: string,
-  options?: any,
-): Promise<never> {
+async function analyzeStatically(targetImage: string, options?: any) {
   const staticScanningOptions = getStaticAnalysisOptions(options);
-  return analyzer.analyzeStatically(targetImage, staticScanningOptions);
+
+  // Relevant only if using a Docker runtime. Optional, but we may consider what to put here
+  // to present to the user in Snyk UI.
+  const runtime = undefined;
+  // Both the analysis and the manifest files are relevant if inspecting a Dockerfile.
+  // This is not the case for static scanning.
+  const dockerfileAnalysis = undefined;
+  const manifestFiles = [];
+
+  try {
+    const staticAnalysis = await analyzer.analyzeStatically(
+      targetImage,
+      staticScanningOptions,
+    );
+
+    const parsedAnalysisResult = parseAnalysisResults(
+      targetImage,
+      staticAnalysis,
+      dockerfileAnalysis,
+    );
+
+    const dependenciesTree = await buildTree(
+      targetImage,
+      parsedAnalysisResult.type,
+      parsedAnalysisResult.depInfosList,
+      parsedAnalysisResult.targetOS,
+    );
+
+    const analysis = {
+      package: dependenciesTree,
+      packageManager: parsedAnalysisResult.type,
+      imageId: parsedAnalysisResult.imageId,
+      binaries: parsedAnalysisResult.binaries,
+      imageLayers: parsedAnalysisResult.imageLayers,
+    };
+
+    return buildResponse(
+      runtime,
+      analysis,
+      dockerfileAnalysis,
+      manifestFiles,
+      staticScanningOptions,
+    );
+  } catch (error) {
+    const analysisError = tryGetAnalysisError(error, targetImage);
+    throw analysisError;
+  }
+}
+
+function tryGetAnalysisError(error, targetImage): Error {
```

Comment:
In order to reflect specific things that may go wrong, I would like to suggest instead creating a custom `AnalysisError` which inherit from JS Error class. coming from .NET and OOP approach, it sounds a bit more extendable. up to you :)

Reply:
I only wanted to abstract out the repeating piece of code, I also agree it's not ideal with the way the implementation either throws or returns an error for the caller to throw, but I've been trying not to touch/improve on the existing code while adding static analysis, which is the focus of the PR 😄Most of the bigger improvements I'm happy to introduce in a separate PR, there's also one other big task pending from a previous review...

Reply:
since this change was introduced in one `feat` commit, It was a bit hard to detect which part of the code is new and which was a part of the existing code. so I reviewed it as a whole. Usually when I see such cases, if it's easy and quick to fix, I try to provide a small improvement. I'll add a code snippet down below, and if you'll find it useful you could use it as well :)